### DISCUSSION TITLE
pytest/pytest-3 name convention

### DISCUSSION THREAD

**Discussion by nbehrnd** [pytest-dev/pytest#9481] [Q&A] [answered]
For `pylint` as provided in for Linux Xubuntu 18.04 LTS, there are/were a separate version addressing Python2 (then called `pylint`, ([tracker](https://packages.ubuntu.com/search?keywords=pylint&searchon=names)) and an other addressing Python3  (then called `pylint3`, [tracker](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=pylint3&searchon=names)).  With the depart of Python2, Xubuntu 20.04 LTS' default is Python3; the linter *for Python3 only* now (again) is called and addressed from the CLI by `pylint`.

To test Python, the same repositories offer `python-pytest` ([tracker](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=python-pytest&searchon=names)), and `python3-pytest` ([tracker](https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=python3-pytest&searchon=names)).  When booting the «live USB drive» version of Xubuntu 20.04.2 LTS, despite the *absence* of Python2, as by tab-completion of `pyt` on the CLI, the checker replies to `pytest-3`; this contrasts to the recently updated documentation of pytest (pdf by Jan 04, 2022) adopting `pytest`.

My questions: Is there a reason the pytest project may influence why the call signs differ?  Or, is it up to the publishers of Linux distributions how to invoke pytest from the CLI, (maybe, additional definition of an appropriate alias) to match the pattern in pytest's handbook?

The attached .zip contains three screen photos.

[differences.zip](https://github.com/pytest-dev/pytest/files/7814667/differences.zip)

**Comment by RonnyPfannschmidt** [comment, marked as answer]
the different name is a choice by the distributions - pytest itself ships the declaration as `pytest` if its different in a distro, then thats on the distro

**Reply by The-Compiler** [reply]
Ideally recent Debian/Ubuntu versions would just point `pytest` to the Python 3 pytest already... but apparently that's not the case: [#970643 - "pytest" command is missing - Debian Bug report logs](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=970643)

**Reply by nbehrnd** [reply]
In the installation I'm using right now (Debian 12/bookworm, branch testing), the tab-completion of `pyt` lists both `pytest` (which is pytest 4.6.11 for Python 2.7.18) and `pytest-3` (which is pytest 6.2.4 for Python 3.9.9).

So far, I assume this co-existence of the two Python interpreters and their corresponding checkers is caused by a continuous update/upgrade of this installation when development for Python2 ended.  Supporting evidences for this line of thought are
1) Debian's [tracker](https://tracker.debian.org/pkg/pytest) replying to `pytest`.  The search with `python-pytest` or `python3-pytest` as shown in the screen photos submitted (synaptic in Xubuntu 20.04.2 LTS Fossa) in this database equally lead to this very landing page.
2) In November 2021, a colleague of mine did a clean installation from zero of Xubuntu 20.04.3 LTS Fossa. By default, he encounters Python3 only, and pytest for Python3 replies (again) to the old `pytest` command.
